code/game/machinery/floodlight.dm

```diff
@@ -1,128 +1,108 @@
 /obj/machinery/floodlight
-	name = "Emergency Floodlight"
-	desc = "Let there be light!"
+	name = "emergency floodlight"
+	desc = "A high-intensity flood lamp on a wheeled platform. It runs on a replaceable power cell."
 	icon = 'icons/obj/machines/floodlight.dmi'
-	icon_state = "flood00"
-	density = 1
-	var/on = 0
-	var/obj/item/cell/cell = null
-	var/use = 200 // 200W light
-	var/unlocked = 0
-	var/open = 0
-	var/brightness_on = 8		//can't remember what the maxed out value is
+	icon_state = "floodlight"
+	density = TRUE
+	active_power_usage = 200
+	var/panel_locked = TRUE
+	var/brightness_on = 8
+	var/datum/component/battery_power/BP
 
 /obj/machinery/floodlight/Initialize()
 	. = ..()
-	cell = new(src)
+	BP = AddComponent(/datum/component/battery_power)
+	if (BP)
+		BP.cell = new(src)
```

Comment:
This will 100% need an argument for the type of cell, because we've got a smattering of objects that spawn with particular cells.

Reply:
Also, is the object responsible for spawning its cell or should the component be?
Also also, do we want a getter function for the cell? Even though the cell _lives_ in the object, it's a _var_ in the component, which feels... just slightly obtuse.

Reply:
it could go either way. do you have any particular preference?